### PR TITLE
feature: integrate OpenVINO AI plugins

### PR DESCRIPTION
**Pull request by frenchwr**
## Background on plugins

This PR integrates OpenVINO AI plugins from from a [repo maintained by Intel](https://github.com/intel/openvino-ai-plugins-gimp). The plugins include stable diffusion, semantic segmentation, and super resolution, and use Intel's AI inference library OpenVINO which supports accelerators like Intel GPUs and NPUs (neural processing units). The NPUs are built into Intel's line of Core Ultra chips that are present in many new laptop and desktop devices, so the hardware support for the plugins, while not universal, is expected to reach a large number of users. The plugins do not yet support GIMP v3, although it's on the roadmap.

## Technical details

The primary changes include integrating with multiple content producer snaps to minimize the maintenance burden on the GIMP snap itself. In summary:

* Add a new plug for `openvino-ai-plugins-gimp` snap (see [PR here](https://github.com/canonical/openvino-ai-plugins-gimp-snap/pull/1) for background) and add a `command-chain` script from this repo to set the runtime environment for GIMP. An update to GIMP's plug-in search path allows these plugins to be used over this interface.
* Add a new plug for `openvino-toolkit-2404` snap ([repo](https://github.com/canonical/openvino-toolkit-snap)) and add a `command-chain` script from this repo to set the runtime environment for GIMP.
* Add a new plug for accessing the Intel NPU char device node on the host over the `custom-device` snapd interface. The slot is provided by [the `intel-npu-driver` snap](https://github.com/canonical/intel-npu-driver-snap/).
* Add a new plug for mounting the NPU runtime libs from [the `intel-npu-driver` snap](https://github.com/canonical/intel-npu-driver-snap/).
* Add `override-stage` section to the gimp part of the snap to update the plugin search path and to fix GIMP's python interpreters file, which was pointing at a python interpreter that does not exist inside the snap.

## Testing

I've tested on a remote machine equipped with an Intel CPU, GPU, and NPU. I have verified that all three AI plugins function on the different hardware device types, with the GPU and NPU providing acceleration.

### CODE REVIEW

**Comment by jnsgruk** [comment]
Changes looking good overall, do you have an idea when GIMP 3.0 support is targeted?

**Comment by frenchwr** [comment]
> Changes looking good overall, do you have an idea when GIMP 3.0 support is targeted?

I asked here: https://github.com/intel/openvino-ai-plugins-gimp/issues/154 

It sounds like it's on their roadmap but they have not committed to any specific date.

**Comment by jnsgruk** [comment]
Okay, might need to consider releasing this to a different track when it's ready for release rather than delay releasing 3.0 when it's out.

**Comment by frenchwr** [comment]
> Okay, might need to consider releasing this to a different track when it's ready for release rather than delay releasing 3.0 when it's out.

That sounds great to me! Is there something I can do (e.g. submit a request for a new track in the store) to help facilitate that? This is something we're hoping to publish and deliver for Intel this quarter.

**Comment by jnsgruk** [comment]
https://forum.snapcraft.io/t/track-request-for-gimp/44274

Once we have the new track, I can help with the branch/credential plumbing here.

**Comment by frenchwr** [comment]
Amazing! Thanks so much @jnsgruk .

**Comment by frenchwr** [comment]
@jnsgruk I went ahead and put in a request for adding autoconnections for the new plugs to the snap here: https://forum.snapcraft.io/t/interface-autoconnections-request-for-gimp-track-2-99-openvino/44296

Assuming everything looks good, would you mind dropping a +1 in that thread since you're the snap maintainer?

**Comment by lucyllewy** [comment]
It's important to note for Canonicalers that the snapcrtafters community do not have access to your internal Jira, so there's not much we can gain from the Jira link being put right at the top of a PR indicating it as the most important part of the comment.

**Comment by frenchwr** [comment]
> It's important to note for Canonicalers that the snapcrtafters community do not have access to your internal Jira, so there's not much we can gain from the Jira link being put right at the top of a PR indicating it as the most important part of the comment.

@lucyllewy Point taken, I removed the link. :) I'm in the habit of dropping link in the PR description because it enables some automation with Jira, but that only works for repos under the Canonical org.

**Comment by frenchwr** [comment]
@jnsgruk Looks like the new track in the store was created ([link](https://forum.snapcraft.io/t/track-request-for-gimp/44274/2))! :partying_face: 

Let me know if there is anything I can do to help publish to the new track.

**Comment by jnsgruk** [comment]
Heyo 👋🏻 I've set up the CI environment and created the `2.99-openvino` branch in this repo, and retargeted this PR so it's merging into that branch.

I also pushed some changes to the CI so that this branch is all wired up correctly. If you're happy to mark this as ready, I think we're good to go from a plumbing perspective 🙂

**Comment by jnsgruk** [comment]
One thing I haven't done is wire up the auto-updater version bump thingy for this track - I think we should wait until there is 3.0 support for that and just merge into the stable track at that point.

**Comment by frenchwr** [comment]
Thanks, @jnsgruk ! Just marked it as ready for review.

File: README.md [hunk 1]
```diff
@@ -78,6 +78,62 @@ Now that your git metadata has been updated you are ready to create a bugfix bra
...
+
+1. Build and install the OpenVINO-enabled GIMP snap locally as it is not currently published:
+
+    ```shell
+    cd gimp
+    snapcraft
+    sudo snap install --dangerous ./gimp_2.99.16_amd64.snap
+    ```
+
+    **TODO**: publish the snap under a new track in the store (e.g. `sudo snap install gimp --channel=openvino-ai-plugins/beta`).
```
Comment (jnsgruk) [inline README.md]: I think this needs updating?

File: README.md [hunk 2]
```diff
@@ -78,6 +78,62 @@ Now that your git metadata has been updated you are ready to create a bugfix bra
...
+3. Connect snap interfaces:
+
+    ```shell
+    sudo snap connect gimp:intel-npu intel-npu-driver:intel-npu
+    sudo snap connect gimp:npu-libs intel-npu-driver:npu-libs
+    sudo snap connect gimp:openvino-libs openvino-toolkit-2404:openvino-libs
+    sudo snap connect gimp:openvino-ai-plugins-gimp-libs openvino-ai-plugins-gimp:openvino-ai-plugins-gimp-libs
+    ```
+
+    **TODO**: request autoconnections from the Snap Store.
```
Comment (jnsgruk) [inline README.md]: I think we can remove this note.

**Review by jnsgruk — COMMENTED** [review]
Two minor things in README then I think I'll merge and we'll see how  the release goes!

File: README.md (hunk 1 above)
Comment (frenchwr) [reply README.md]: Good call! Done.

File: README.md (hunk 2 above)
Comment (frenchwr) [reply README.md]: Thanks, done.

**Comment by frenchwr** [comment]
> Two minor things in README then I think I'll merge and we'll see how the release goes!

Glad you caught that as there were a few other sections of the README that needed to be tweaked as well. I left the interface connection blocks for now but hopefully those can be removed soon once the autoconnections are approved.

**Review by jnsgruk — APPROVED** [review]

**Comment by jnsgruk** [comment]
Keep an eye out here! https://github.com/snapcrafters/gimp/actions/runs/12377350056

If we're successful, there should be a Call For Testing issue opened automatically once it's built.

**Pull request merged**